homeassistant/components/subaru/__init__.py

```diff
@@ -0,0 +1,192 @@
+    UPDATE_INTERVAL,
+    VEHICLE_API_GEN,
+    VEHICLE_HAS_EV,
+    VEHICLE_HAS_REMOTE_SERVICE,
+    VEHICLE_HAS_REMOTE_START,
+    VEHICLE_HAS_SAFETY_SERVICE,
+    VEHICLE_LAST_UPDATE,
+    VEHICLE_NAME,
+    VEHICLE_VIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+REMOTE_SERVICE_SCHEMA = vol.Schema({vol.Required(VEHICLE_VIN): cv.string})
+
+
+async def async_setup(hass, base_config):
+    """Do nothing since this integration does not support configuration.yml setup."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Subaru from a config entry."""
+    config = entry.data
+    websession = aiohttp_client.async_get_clientsession(hass)
+    try:
+        controller = SubaruAPI(
+            websession,
+            config[CONF_USERNAME],
+            config[CONF_PASSWORD],
+            config[CONF_DEVICE_ID],
+            config[CONF_PIN],
+            None,
+            config[CONF_COUNTRY],
+            update_interval=UPDATE_INTERVAL,
+            fetch_interval=FETCH_INTERVAL,
+        )
+        _LOGGER.debug("Using subarulink %s", controller.version)
+        await controller.connect()
+    except InvalidCredentials:
+        _LOGGER.error("Invalid account")
+        return False
+    except SubaruException as err:
+        raise ConfigEntryNotReady(err.message) from err
+
+    vehicle_info = {}
+    for vin in controller.get_vehicles():
+        vehicle_info[vin] = get_vehicle_info(controller, vin)
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            return await refresh_subaru_data(entry, vehicle_info, controller)
+        except SubaruException as err:
+            raise UpdateFailed(err.message) from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=COORDINATOR_NAME,
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=FETCH_INTERVAL),
+    )
+
+    await coordinator.async_refresh()
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        ENTRY_CONTROLLER: controller,
+        ENTRY_COORDINATOR: coordinator,
+        ENTRY_VEHICLES: vehicle_info,
+    }
+
+    for component in SUPPORTED_PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    async def async_fetch_service(call):
+        """Force refresh of data from Subaru."""
+        vin = call.data[VEHICLE_VIN].upper()
+        if vin not in vehicle_info.keys():
+            hass.components.persistent_notification.async_create(
+                f"ERROR - Invalid VIN provided while calling {call.service}", "Subaru"
+            )
+        else:
+            if call.service == REMOTE_SERVICE_FETCH:
+                await coordinator.async_refresh()
+
+    hass.services.async_register(
+        DOMAIN, REMOTE_SERVICE_FETCH, async_fetch_service, schema=REMOTE_SERVICE_SCHEMA
+    )
+
```

Comment:
```suggestion
```

We actually already have a service that does this `homeassistant.update_entity`: https://www.home-assistant.io/docs/scripts/service-calls/#homeassistant-services

We shouldn't reimplement it.   Since its using DataUpdateCoorinator, calling it on any entity will update them all.